bin/alias.ml

```diff
@@ -79,14 +79,40 @@ let dep_on_alias_rec_multi_contexts ~dir:src_dir ~name ~contexts =
   let* dir =
     Action_builder.of_memo (find_dir_specified_on_command_line ~dir:src_dir)
   in
-  let+ is_nonempty_list =
+  let* is_nonempty_list =
     Action_builder.all
       (List.map contexts ~f:(fun ctx ->
            Action_builder.dep_on_alias_rec name ctx dir))
   in
+  let+ load_dir =
```

Comment:
would it make sense to load the dir only if we're in the error case?

Reply:
I'm not super sure about the semantics here. I need to do it in the monad, but I would like this to be lazy or something at least.

Reply:
Nervermind I am talking nonsense. I worked out how to do it.